src/wallet/wallet2.cpp

```diff
@@ -1206,8 +1206,22 @@ void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& re
   get_short_chain_history(short_chain_history);
   m_run.store(true, std::memory_order_relaxed);
   if (start_height > m_blockchain.size() || m_refresh_from_block_height > m_blockchain.size()) {
-    if (!start_height)
+    if (!start_height) {
+
+      std::string daemon_height_err = "";
+      uint64_t daemon_bc_height = this->get_daemon_blockchain_height(daemon_height_err);
```

Comment:
Thanks for pointing out the pitfalls. I attempted to address these in the latest commit. Do you think it's better to just set the restore height to target if it's way high, or actually stick to the user specified height? The latest commit just goes with user height no matter how far in the future.